app/models/compute_resources/foreman/model/ec2.rb

```diff
@@ -87,6 +92,14 @@ def zones
     end
     alias_method :available_zones, :zones
 
+    def root_device_from_ami(ami_id)
+      ami_desc = client.describe_images('image-id' => ami_id).body["imagesSet"][0]
+      root_device_name = ami_desc["rootDeviceName"]
```

Comment:
@eLvErDe, 
While testing the changes I observed couple of things:
 - Many images have a empty value for `rootDeviceName` so need to handle this condition here.

- For images with root device name, I didn't get `volumeSize` key for it under `blockDeviceMapping`. Along with device name, I can see key called `virtualName`. Example - `{"blockDeviceMapping"=> [{"deviceName"=>"sdb", "virtualName"=>"ephemeral0"}]`

Any thoughts on this?

Reply:
In case of EBS volume, it contains "volumeSize".

Reply:
Hello,

Sadly I'm not sure about that...

Reply:
Added condition which only add boot device mapping when device type from image is EBS.
https://github.com/EarthLab-Luxembourg/foreman/pull/5

Reply:
Thank you @eLvErDe! Tests are not happy so could you please do rebase?

Reply:
Just did, I also did for PR#2170